compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -413,12 +413,24 @@ impl SirFuncCx<'tcx> {
             mir::Rvalue::Cast(mir::CastKind::Misc, op, ty) => self.lower_cast_misc(bx, bb, op, ty),
             mir::Rvalue::Len(p) => {
                 let ip = self.lower_place(bx, bb, p);
-                self.offset_iplace(
-                    bx,
-                    ip,
-                    i32::try_from(self.tcx.data_layout.pointer_size.bits()).unwrap(),
-                    dest_ty,
-                )
+                match p.ty(&self.mir.local_decls, self.tcx).ty.kind() {
+                    ty::Array(_elem_ty, len) => {
+                        let raw_val =
+                            usize::try_from(len.eval_usize(self.tcx, ty::ParamEnv::reveal_all()))
```

Comment:
Does `eval_usize` return a `usize`? [The name is suggestive!]

Reply:
They suckered you in too!

https://github.com/rust-lang/rust/blob/b5c496de374407c0937cf5257d89230b5358d053/compiler/rustc_middle/src/ty/consts.rs#L197-L202

Reply:
I was going to write something sarky, but life is too short!

Reply:
If it makes you feel better, this confused me too!